### ISSUE TITLE
Adding a mixed option to QuantityFactory

### ISSUE THREAD

**Issue by keilw** [unitsofmeasurement/unit-api#187]
In addition to https://github.com/unitsofmeasurement/unit-api/issues/185 QuantityFactory may be extended to allow the equivalent of [Wolfram](https://reference.wolfram.com/language/ref/MixedUnit.html)'s `Quantity[MixedMagnitude[{1, 2}], MixedUnit[{"Feet", "Inches"}]]` Overriding the `create()` method passing an array of numbers in addition to a single `Number`.

Needs #185

**Comment by dautelle** [comment]
Method name Unit.with(Unit)

**Comment by thodorisbais** [comment]
+1 again, from my side. @dautelle there is already a `create()` method which will be overridden.

**Comment by keilw** [comment]
Yes, that's the idea, but this depends on #185. The earlier pointer to MapleSoft showed there is a method `Unit.combine()` https://www.maplesoft.com/support/help/Maple/view.aspx?path=combine%2funits but it looks like a method that does all sorts of things, practically a combination of most we defined here like alternate, transform or the mix/compound one.

@dautelle I'm afraid `Unit.with(Unit)` does not say anything what it does, or do others get the meaning if they just read the method signature in an IDE or elsewhere? The problem with combine() is also similar to compound() although it seems a little less obvious that a "combined unit" may not be as obvious to be a "derived" 
Unit.with() is really just unclear. The only place that is used would be some Builder Pattern implementations instead of `setX()` or `setY()`. `java.time` does also have some withYX() methods, but most contain the purpose in the method name like `withChronology()` etc. The only exceptions are `with(Temporal)`, `with(TemporalAdjuster)` and `with(TemporalField)` but adding this for a simple case beside totally different patterns like `prefix(Prefix)` or `transform(UnitConverter)` sounds weird and inconsequent. We cannot simply change transform() because it's part of the 1.0 API but if a majority would prefer a series of `with()` methods like JSR 310 like
```
Unit<Q> with(UnitConverter converter);
Unit<Q> with(Prefix prefix);
Unit<Q> with(Unit<Q> that);
```
Then I think I could find that reasonable and consequent. transform(UnitConverter) shold be deprecated till a new version can remove it.

WDYT?

**Comment by desruisseaux** [comment]
What would be the difference (except formatting) between "2.5 feet" and "2 feet 6 inch"?

**Comment by keilw** [comment]
The letter completely loses the knowledge of the inch. That's the point to model and state the exact units in a place, and preserve them so a quantity may also be constructed accordingly.

**Comment by desruisseaux** [comment]
> The later completely loses the knowledge of the inch.

Given that the 2.5 value can be separated to feet in inches as below, can you elaborate on which information is lost?

```
double value = 2.5;
double feet = Math.floor(value);        // 2 feet
double inch = (value - feet) * 12;      // 6 inches
```

**Comment by keilw** [comment]
Closing this in favor of #190
